pkg/manifest/manifest.go

```diff
@@ -5,38 +5,36 @@
 package manifest
 
 import (
+	"encoding"
 	"errors"
 	"net/http"
 
 	"github.com/ethersphere/bee/pkg/swarm"
 )
 
+// ErrNotFound is returned when an Entry is not found in the manifest
 var ErrNotFound = errors.New("manifest: not found")
 
-// Parser for manifest
-type Parser interface {
-	// Parse parses the encoded manifest data and returns the result
-	Parse(bytes []byte) (Interface, error)
-}
-
 // Interface for operations with manifest
 type Interface interface {
-	// Add a manifest entry to specified path
+	// Add a manifest entry to the specified path
 	Add(string, Entry)
-	// Remove reference from file on specified path
+	// Remove a manifest entry on the specified path
 	Remove(string)
-	// FindEntry returns manifest entry if one is found on specified path
+	// FindEntry returns a manifest entry if one is found in the specified path
 	FindEntry(string) (Entry, error)
-	// Serialize return encoded manifest
-	Serialize() ([]byte, error)
+	// BinaryMarshaler is the interface implemented by an object that can marshal itself into a binary form
+	encoding.BinaryMarshaler
+	// BinaryUnmarshaler is the interface implemented by an object that can unmarshal a binary representation of itself
+	encoding.BinaryUnmarshaler
 }
 
-// Entry represents single manifest entry
+// Entry represents a single manifest entry
 type Entry interface {
-	// GetReference returns address of the entry file
+	// GetReference returns the address of the file in the entry
 	GetReference() swarm.Address
-	// GetName returns the name of the file for the entry, if added
+	// GetName returns the name of the file in the entry
 	GetName() string
-	// GetHeaders returns the headers for manifest entry, if configured
+	// GetHeaders returns the headers for the file in the manifest entry
```

Comment:
I'm not really sure why we need to tie the manifest interface definition to the `http` package. This rapes manifests into an opinionated implementation which deviates from the generalisation it is supposed to provide. This is not a blocker, but the business logic on which headers to return to the requester need to live in the http api package, not inside manifests.